lib/mumuki/domain/helpers/user.rb

```diff
@@ -87,8 +87,21 @@ def main_organization
     student_granted_organizations.first || any_granted_organizations.first
   end
 
+  # Deprecated: use `immersive_organization_at` which
+  # properly looks for a single immersive organization taking
+  # current organization and path into account
   def has_immersive_main_organization?
-    !!main_organization.try(&:immersive?)
+    main_organization.try { |it| it if it.immersive? }.present?
+  end
+
+  def immersive_organization_at(path_item, current = Organization.current)
```

Comment:
In this PR I don't see where path_item could be nil. What other cases are you considering?